### PR TITLE
Add ContainerRuntimeLoader

### PR DESCRIPTION
**Pull request by chalasr**
Next to https://github.com/symfony/symfony/pull/21625#discussion_r101715445

### CODE REVIEW

File: lib/Twig/ContainerRuntimeLoader.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    public function load($class)
+    {
+        if ($this->container->has($class)) {
+            return $this->container->get($class);
+        }
+
+        if (null !== $this->logger) {
+            $this->logger->warning(sprintf('Class "%s" is not configured as a Twig runtime service.', $class));
```
Comment (stof) [inline lib/Twig/ContainerRuntimeLoader.php]: I'm not sure we should keep this logging call. If another loader is able to load the runtime, this is just spamming logs (especially at warning level). and if no loader can load it, ``Twig_Environment::getRuntime`` throws an exception with a meaningful message already

Comment (chalasr) [inline lib/Twig/ContainerRuntimeLoader.php]: Indeed, removed the logger dependency.

File: lib/Twig/ContainerRuntimeLoader.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Psr\Container\ContainerInterface;
+
+/**
+ * Lazily loads Twig runtime implementations from a container.
```
Comment (stof) [inline lib/Twig/ContainerRuntimeLoader.php]: I would say ``from a PSR-11 container``.

And the long description of the class should explain that it expects being able to use the runtime class as an identifier in the container

Comment (chalasr) [inline lib/Twig/ContainerRuntimeLoader.php]: done

**Comment by fabpot** [comment]
Should be for the 1.x branch in which we still can add new features (as long as they don't break BC of course as we cannot add new deprecation notices).

**Comment by chalasr** [comment]
Rebased!

File: CHANGELOG [hunk 3]
```diff
@@ -1,4 +1,8 @@
-* 1.31.0 (2017-XX-XX)
+* 1.31.1 (2017-XX-XX)
```
Comment (fabpot) [inline CHANGELOG]: That's a new feature, so we need to bump to 1.32

**Comment by fabpot** [comment]
I will take care of bumping the version when merging.

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**